### PR TITLE
Rashmil dev

### PR DESCRIPTION
**Pull request by Rashmil-1999**
Please review the work and let me know of further changes.

### CODE REVIEW

File: app/api/api_v1/endpoints/login.py [hunk 1]
```diff
@@ -43,16 +61,41 @@ def login_access_token(
...
+
+    Parameters
+    ----------
+    current_user : models.User, optional
+        Instance of the User who is currently logged in, by default Depends(deps.get_current_user)
+
+    Returns
+    -------
+    Any
+        [description]
```
Comment (ka7eh) [inline app/api/api_v1/endpoints/login.py:74]: Looks like this didn't get updated! :)

File: app/api/api_v1/endpoints/login.py [hunk 2]
```diff
@@ -74,8 +117,30 @@ def reset_password(
...
+        The database session, by default Depends(deps.get_db)
+
+    Returns
+    -------
+    Any
+        Message confirmation for password reset.
+
+    Raises
+    ------
+    HTTPException
```
Comment (ka7eh) [inline app/api/api_v1/endpoints/login.py:138]: We can aggregate exceptions of same type and just list the different situations such exception might happen. E.g.:

```
HTTPException
    Raised when the user is not found in the system or is inactive.
```

File: app/api/api_v1/endpoints/users.py [hunk 3]
```diff
@@ -94,8 +157,30 @@ def create_user_open(
...
+
+    Returns
+    -------
+    Any
+        Instance of the created User.
+
+    Raises
+    ------
+    HTTPException
+        [description]
```
Comment (ka7eh) [inline app/api/api_v1/endpoints/users.py:181]: Looks like these are left out too.

File: app/core/config.py [hunk 4]
```diff
@@ -10,6 +10,16 @@
 
 
 class Settings(BaseSettings):
+    """Settings is class to hold all the configuration information about the server
+
+    Parameters
```
Comment (ka7eh) [inline app/core/config.py:15]: This part is not needed. `BaseSettings` is just the parent class and shouldn't be documented as a parameter.

**Review by ka7eh — COMMENTED** [review]
Looking good. I added a few comments, some of them may apply to multiple places.

**Comment by Rashmil-1999** [comment]
I just made the changes. I believe it's pretty much consistent now. Let me know if there is something left out.

File: app/crud/base.py [hunk 5]
```diff
@@ -36,7 +36,7 @@ def get(self, db: Session, id: Any) -> Optional[ModelType]:
 
         Returns
         -------
-        Optional[ModelType]
+        Optional
```
Comment (ka7eh) [inline app/crud/base.py:39]: This was correct before (`Optional[ModelType]`). Same thing for the new changes in this section.

Comment (Rashmil-1999) [reply app/crud/base.py:39]: Let me correct it and I'll make a new commit.

File: .gitignore [hunk 6]
```diff
@@ -6,6 +6,8 @@ __pycache__/
 .coverage
 htmlcov
 .vscode/
+.pytest_cache
```
Comment (ka7eh) [inline .gitignore:9]: Not necessary to update for now, but these are not needed. These two folders have their own `.gitignore` inside them which exclude everything in them.

**Review by ka7eh — APPROVED** [review]

**Pull request merged**